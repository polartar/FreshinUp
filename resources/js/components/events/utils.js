import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
