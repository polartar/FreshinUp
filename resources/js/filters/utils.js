export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
