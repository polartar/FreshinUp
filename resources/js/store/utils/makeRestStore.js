/**
 * All of this below is just so that when we call store.dispatch('module/deleteItem', { getItems: true })
 * it will get them with the needed include if that was set previously
 *
 * Example:
 * From
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 * store.dispatch('module/deleteItem', { getItems: false, id: item.uuid })
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 *
 * To
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 * store.dispatch('module/deleteItem', { getItems: true, id: item.uuid })
 *
 * Hey that's one line less and a lot of code duplication avoided for each module
 */

import mapKeys from 'lodash/mapKeys'
import {
  _getPaginationAsParams, _getSortingAsParams,
  buildApi,
  makeModule as _makeModule
} from '@freshinup/core-ui/src/store/utils/makeRestStore'
import pickBy from 'lodash/pickBy'
import omitBy from 'lodash/omitBy'

export const makeModule = (store, moduleName = '') => {
  const m = _makeModule(store, moduleName)
  const _actions = store.actions

  m.mutations = {
    ...m.mutations,
    setFilters: (state, payload) => {
      payload = omitBy(payload, (value, key) => {
        return key.indexOf('sort') > -1
      })
      state.filters = mapKeys(payload, (value, key) => {
        if (['include'].includes(key)) {
          return key
        }
        return key.replace(/filter\[(.*)]/gi, '$1')
      })
    }
  }
  m.actions = {
    ...m.actions,
    getItems (context, payload = { params: {}, data: {} }) {
      const pagination = context.getters.pagination
      const sorting = context.getters.sorting
      // Convert filters into query parameters that is JSON API compliant
      const filters = mapKeys(context.state.filters, function (value, key) {
        // In order to be backwards compatible we need to ignore a few keys
        //    No deprecation plan yet as there is not a design change planned.
        //    Most likely we'll need a setSort() and term may remain
        if (['term', 'q', 'sort', 'include'].indexOf(key) > -1) {
          return key
        }
        return key.indexOf('filter[') === 0 ? key : `filter[${key}]`
      })

      // Deprecation Notice
      const deprecatedKeys = Object.keys(pickBy(payload.params, (value, key) => {
        return key.indexOf('filter[') > -1 || key.indexOf('sort') > -1
      }))
      if (deprecatedKeys.length > 0) {
        console.warn(`
FreshBUS: Version 1.15 of FreshBUS is the last support filtering through getList({ params }) on Store Modules made from makeRestStore.
Please switch to using setFilters() before calling getItems(). Keys provided are ${deprecatedKeys} on ${moduleName}
          `)
      }
      payload.params = {
        ..._getPaginationAsParams(pagination),
        ..._getSortingAsParams(sorting),
        ...filters,
        ...payload.params
      }

      return _actions.getItems.apply(null, [context, payload])
    }
  }
  return m
}

/**
 * Quickly build a REST Module with our Fresh Standards
 * @param name
 * @param items Initial State for Items
 * @param item  Initial State for Item (selected item)
 * @param filters
 * @param basePaths
 * @param options
 */
export default (name, { items, item, filters }, basePaths, options = {}) => {
  const api = buildApi(name, { items, item, filters }, basePaths, options)
  return makeModule(api.getStore(), name)
}
