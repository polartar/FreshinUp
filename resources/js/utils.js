import get from 'lodash/get'
/**
 *
 * @param options HTML attributes
 * @param [options.type] string 'link'|'script' or anything in fact
 * @param [options.container] HTMLElement
 * @returns {Promise<unknown>}
 */
export const lazyLoad = (options) => {
  return new Promise((resolve, reject) => {
    if (!options.type) {
      return reject(new Error('options.type is required'))
    }
    if (!options.container) {
      options.container = document.body
    }
    const { type, container, ...attributes } = options
    let element = document.createElement(type)
    Object.keys(attributes).forEach(key => element.setAttribute(key, attributes[key]))
    element.addEventListener('load', () => {
      resolve(element)
    })
    element.addEventListener('error', (e) => {
      reject(e)
    })
    container.append(element)
  })
}

export const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}
