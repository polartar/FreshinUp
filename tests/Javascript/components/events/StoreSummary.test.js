import { shallowMount } from '@vue/test-utils'
import { createLocalVue } from 'fresh-bus/tests/utils'
import Component from '~/components/events/StoreSummary.vue'
import { FIXTURE_EVENT_STORE_SUMMARY } from 'tests/__data__/storeSummary'

describe('Event StoreSummary component', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
    })
    test('required set', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          store: FIXTURE_EVENT_STORE_SUMMARY
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
    })
    test('onButtonClick() emits onButtonClick', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          store: FIXTURE_EVENT_STORE_SUMMARY
        }
      })
      wrapper.vm.onButtonClick()
      expect(wrapper.emitted()['onButtonClick']).toBeTruthy()
    })
    test('onRemoveClick() emits onRemoveClick', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          store: FIXTURE_EVENT_STORE_SUMMARY
        }
      })
      wrapper.vm.onRemoveClick()
      expect(wrapper.emitted()['onRemoveClick']).toBeTruthy()
    })
  })
})
