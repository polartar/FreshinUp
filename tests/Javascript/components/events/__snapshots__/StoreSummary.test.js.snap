// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event StoreSummary component Snapshots required set 1`] = `
<v-card-stub
  tag="div"
>
  <v-container-stub
    font-weight-bold=""
    grey--text=""
    py-3=""
    subheading=""
    tag="div"
  >
    
    Fleet Member Summary
  
  </v-container-stub>
   
  <v-divider-stub />
   
  <v-container-stub
    font-weight-bold=""
    grey--text=""
    pt-0=""
    tag="div"
  >
    <v-layout-stub
      align-center=""
      py-2=""
      tag="div"
    >
      <v-flex-stub
        caption=""
        font-weight-bold=""
        tag="div"
      >
        
        OWNER
      
      </v-flex-stub>
       
      <v-flex-stub
        subheading=""
        tag="div"
        text-truncate=""
        text-xs-right=""
        xs6=""
      >
        
        Dan Smith
      
      </v-flex-stub>
    </v-layout-stub>
     
    <v-divider-stub />
     
    <v-layout-stub
      align-center=""
      py-2=""
      tag="div"
    >
      <v-flex-stub
        caption=""
        font-weight-bold=""
        tag="div"
      >
        
        LISENCE DUE
      
      </v-flex-stub>
       
      <v-flex-stub
        subheading=""
        tag="div"
        text-truncate=""
        text-xs-right=""
        xs6=""
      >
        
        Dec, 30 2020
      
      </v-flex-stub>
    </v-layout-stub>
     
    <v-divider-stub />
     
    <v-layout-stub
      align-center=""
      py-2=""
      tag="div"
    >
      <v-flex-stub
        caption=""
        font-weight-bold=""
        tag="div"
      >
        
        CONTACT PHONE
      
      </v-flex-stub>
       
      <v-flex-stub
        subheading=""
        tag="div"
        text-truncate=""
        text-xs-right=""
        xs6=""
      >
        
        938 374822
      
      </v-flex-stub>
    </v-layout-stub>
     
    <v-divider-stub />
     
    <v-layout-stub
      align-center=""
      class="py-2"
      tag="div"
    >
      <v-flex-stub
        caption=""
        font-weight-bold=""
        tag="div"
      >
        
        CONTACT EMAIL
      
      </v-flex-stub>
       
      <v-flex-stub
        subheading=""
        tag="div"
        text-truncate=""
        text-xs-right=""
        xs6=""
      >
        
        dan.simth@gmail.com
      
      </v-flex-stub>
    </v-layout-stub>
     
    <v-divider-stub />
     
    <v-layout-stub
      font-weight-bold=""
      grey--text=""
      mt-2=""
      pa-1=""
      tag="div"
      tags_title=""
    >
      <v-flex-stub
        tag="div"
      >
        TAGS
      </v-flex-stub>
    </v-layout-stub>
     
    <v-layout-stub
      row=""
      tag="div"
      wrap=""
    >
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        SEAFOOD
      
      </f-chip-stub>
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        SMOKED
      
      </f-chip-stub>
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        DESSERT
      
      </f-chip-stub>
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        BAY AREA
      
      </f-chip-stub>
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        VEGAN OPTIONS
      
      </f-chip-stub>
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        SMOKED
      
      </f-chip-stub>
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        DESSERT
      
      </f-chip-stub>
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        SEAFOOD
      
      </f-chip-stub>
      <f-chip-stub
        class="my-1 ml-0 mr-2"
        color="primary"
        value="true"
      >
        
        SMOKED
      
      </f-chip-stub>
    </v-layout-stub>
     
    <v-divider-stub
      class="mt-3"
    />
     
    <v-btn-stub
      activeclass="v-btn--active"
      block="true"
      class="grey ma-0 mt-4 white--text text-none"
      tag="button"
      type="button"
    >
      
      View Fleet Member Profile
    
    </v-btn-stub>
     
    <v-btn-stub
      activeclass="v-btn--active"
      block="true"
      class="transparent elevation-0 ma-0 pa-0 caption  grey--text mt-3 button-remove text-none"
      tag="button"
      type="button"
    >
      <v-icon-stub
        class="pr-2 caption"
      >
        
        far fa-trash-alt
      
      </v-icon-stub>
       
      <span>
        Remove fleet member from this event
      </span>
    </v-btn-stub>
  </v-container-stub>
</v-card-stub>
`;
